src/PiWeb.Import.Sdk/IPlugin.cs

```diff
@@ -31,7 +29,32 @@ public interface IPlugin
     /// </summary>
     /// <param name="context">Contains information about the environment.</param>
     /// <exception cref="ModuleRegistrationException">Thrown when module registration fails.</exception>
-    Task Init(IPluginContext context);
+    Task Init(IPluginContext context)
+    {
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Creates a new instance of an <see cref="IImportAutomation"/>
+    /// or <c>null</c> if this plugin does not provide an import automation.
+    /// </summary>
+    /// <param name="context">Contains information about the environment.</param>
+    /// <exception cref="ModuleRegistrationException">Thrown when module registration fails.</exception>
+    IImportAutomation? GetImportAutomation(IImportAutomationContext context)
+    {
+        return null;

```

Comment:
We might want to avoid the nullable return type here. This suggests to the implementer that returning null is a viable option but it it not. We could simply throw an InvalidOperationException in the default implementation and make sure it is not called when not specified in the manifest.

Reply:
Good point. I'll change this to `InvalidOperationException`.

Reply:
Should we also drop `<exception cref="ModuleRegistrationException">Thrown when module registration fails.</exception>` and expect no exception to thrown?

Or even just allow `ModuleNotImplementedException` here, which would be the default implementation, so we can check for it and display specific diagnostic message?

Reply:
Yeah ModuleNotImplementedException with diagnostic message sounds good to me.